File: app/src/main/java/com/rtukpe/fixtures/di/module/ApplicationModule.java
package com.rtukpe.fixtures.di.module;


import android.app.Application;
import android.content.Context;

import com.rtukpe.fixtures.data.manager.AppDataManager;
import com.rtukpe.fixtures.data.manager.DataManager;
import com.rtukpe.fixtures.di.annotations.ApplicationContext;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by rtukpe on 13/03/2018.
 */

@Module
public class ApplicationModule {

    private final Application mApplication;

    public ApplicationModule(Application mApplication) {
        this.mApplication = mApplication;
    }

    @Provides
    @ApplicationContext
    Context provideContext() {
        return mApplication;
    }

    @Provides
    Application provideApplication() {
        return mApplication;
    }

    @Provides
    @Singleton
    DataManager provideDataManager(AppDataManager appDataManager) {
        return appDataManager;
    }
}

File: app/src/main/java/com/rtukpe/fixtures/di/module/ApplicationModule.kt
package com.rtukpe.fixtures.di.module


import android.app.Application
import android.content.Context
import com.rtukpe.fixtures.data.manager.AppDataManager
import com.rtukpe.fixtures.data.manager.DataManager
import com.rtukpe.fixtures.di.annotations.ApplicationContext
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by rtukpe on 13/03/2018.
 */

@Module
class ApplicationModule(private val mApplication: Application) {

    @Provides
    @ApplicationContext
    internal fun provideContext(): Context = mApplication

    @Provides
    internal fun provideApplication(): Application = mApplication

    @Provides
    @Singleton
    internal fun provideDataManager(appDataManager: AppDataManager): DataManager = appDataManager
}
